extension/scraper.js

```diff
@@ -0,0 +1,156 @@
+let firstRun = true;
+let lockPoll = false;
+
+module.exports = function (nodecg) {
+	nodecg.log.info('Polling donations every %d seconds...', POLL_INTERVAL / 1000);
+
+	const extraLifeIdRep = nodecg.Replicant('extralife-id', {defaultValue: ''});
+	const extraLifeTeamIdRep = nodecg.Replicant('extralife-team-id', {defaultValue: ''});
+	const teamGoalRep = nodecg.Replicant('team-goal', {defaultValue: 0});
+	const teamRaisedRep = nodecg.Replicant('team-raised', {defaultValue: 0});
+	const yourGoalRep = nodecg.Replicant('your-goal', {defaultValue: 0});
+	const yourRaisedRep = nodecg.Replicant('your-raised', {defaultValue: 0});
+	const donationsRep = nodecg.Replicant('donations', {defaultValue: {clear: 1, array: []}, persistent: false});
+	const lastSeenDonationRep = nodecg.Replicant('last-seen-donation', {defaultValue: null, persistent: false});
+
+	teamId = extraLifeTeamIdRep.value || null;
+	participantId = extraLifeIdRep.value || null;
+
+	extraLifeIdRep.on('change', function (newValue) {
+		donationsRep.value.clear = donationsRep.value.clear + 1;
+		donationsRep.value.array.length = 0;
+		yourRaisedRep.value = 0;
+		yourGoalRep.value = 0;
+		lastSeenDonationRep.value = null;
+		firstRun = true;
+		participantId = newValue;
+
+		update();
+	});
+
+	extraLifeTeamIdRep.on('change', function (newValue) {
+		donationsRep.value.clear = donationsRep.value.clear + 1;
+		donationsRep.value.array.length = 0;
+		teamRaisedRep.value = 0;
+		teamGoalRep.value = 0;
+		lastSeenDonationRep.value = null;
+		firstRun = true;
+		teamId = newValue;
+
+		update();
+	});
+
+
+	// Get initial data
+	update();
+
+	async function update() {
+		try {
+			if (currentTimeout) {
+				clearTimeout(currentTimeout);
+			}
+
+			if (lockPoll) {
+				return;
+			}
+
+			lockPoll = true;
+			currentTimeout = null;
+
+			if (!firstRun && donationsRep.value.clear) {
+				donationsRep.value.clear = 0;
+			}
+
+			firstRun = false;
+
+			if (!participantId) {
+				currentTimeout = setTimeout(update, POLL_INTERVAL);
+				return;
+			}
+
+			await updateDonations();
+			await updateParticipantTotal();
+
+			if (teamId) {
+				await updateTeamTotal();
+			}
+		} catch (error) {
+			nodecg.log.error('Error scraping Extra Life API:', error);
+		} finally {
+			currentTimeout = setTimeout(update, POLL_INTERVAL);
+			lockPoll = false;
+		}
+	}
+
+	async function updateDonations() {
+		// Note: donations.countDonations is not trustworthy, use from user data instead
+		const donationInfo = await extralifeMock.getUserDonations(participantId);
+
+		// Note: When empty the api returns the donations as an empty string instead of an empty array
+		if (!donationInfo || donationInfo.donations === '') {
+			nodecg.log.error('No donations found for stream ID');
+		}
+
+		const temporary = [];
+		let stop = false;
+
+		donationInfo.donations.forEach(function (donation) {
+			if (stop) {
+				return;
+			}
+
+			donation.id = crypto.createHash('sha256').update(JSON.stringify(donation)).digest('hex');
```

Comment:
`donorID` can probably be used instead of generating an id. I don't know why they chose that name, but apparently its a UUID for the `donation` not for the `donor` so it should be unique to Donation (at least according to the official API docs) https://github.com/DonorDrive/PublicAPI